src/Symfony/Component/Security/Http/Firewall/AccessListener.php

```diff
@@ -47,10 +50,6 @@ public function __construct(TokenStorageInterface $tokenStorage, AccessDecisionM
      */
     public function handle(GetResponseEvent $event)
     {
-        if (null === $token = $this->tokenStorage->getToken()) {
```

Comment:
I think we should keep this block. The exception message is more tailored than the one in `AuthorizationChecker`

Reply:
In my opinion message from `AuthorizationChecker` is better from DX perspective, because it is hinting about why token may be missing.

Reply:
The message in `AuthorizationChecker` reads as

> The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL.

Is it possible to have the AccessListener configured but not firewall configured? If not, then the message would be misleading.

But there is something else: this moves the check for a token after the check for attributes. Is it legitimate?

Reply:
`AccessListener` is configured by https://github.com/symfony/symfony/blob/5e28ac31504e3a7cd2d16b0216182d767f37ad41/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L169

So it's impossible to have configured `AccessListener` without firewalls, unless developers are doing it somehow else.

Reply:
OK to me, lets continue here :)